src/DeviceInterface/InterfacePrototype.js

```diff
@@ -806,7 +806,17 @@ class InterfacePrototype {
                 password: this.passwordGenerator.password,
                 username: this.emailProtection.lastGenerated,
             });
-            this.storeFormData(formData, 'formSubmission');
+
+            // If credentials has only username field, and no password field, then trigger is a partialSave
+            const isUsernameOnly =
+                Boolean(formData.credentials?.username) && !formData.credentials?.password && form.inputs.credentials.size === 1;
```

Comment:
Shouldn't you just check for username and size here? Also, don't need to force to Boolean.

```suggestion
                formData.credentials?.username && form.inputs.credentials.size === 1;
```